light-zk.js/src/merkleTree/merkleTreeConfig.ts

```diff
@@ -47,17 +53,35 @@ export class MerkleTreeConfig {
     this.connection = connection;
   }
 
-  async initializeNewTransactionMerkleTree(
-    oldTransactionMerkleTree: PublicKey,
-    newTransactionMerkleTree: PublicKey,
-  ) {
+  async initializeNewMerkleTrees() {
     if (!this.payer) throw new Error("Payer undefined");
     this.getMerkleTreeAuthorityPda();
+
+    let merkleTreeAuthorityAccountInfo =
+      await this.getMerkleTreeAuthorityAccountInfo();
+    const transactionMerkleTreeIndex =
+      merkleTreeAuthorityAccountInfo.transactionMerkleTreeIndex;
+    const eventMerkleTreeIndex =
+      merkleTreeAuthorityAccountInfo.eventMerkleTreeIndex;
+
+    const oldTransactionMerkleTree =
+      MerkleTreeConfig.getTransactionMerkleTreePda(
+        transactionMerkleTreeIndex.sub(BN_1),
+      );
+    const newTransactionMerkleTree =
+      MerkleTreeConfig.getTransactionMerkleTreePda(transactionMerkleTreeIndex);
+    const oldEventMerkleTree = MerkleTreeConfig.getEventMerkleTreePda(
+      eventMerkleTreeIndex.sub(BN_1),
+    );
+    const newEventMerkleTree =
+      MerkleTreeConfig.getEventMerkleTreePda(eventMerkleTreeIndex);
+
     const tx = await this.merkleTreeProgram.methods
-      .initializeNewTransactionMerkleTree(new anchor.BN("50"))
+      .initializeNewMerkleTrees(new anchor.BN("50"))
```

Comment:
what is the purpose of BN(50)?

Reply:
looked it up, can you add a constant for this for example DEFAULT_MERKLE_TREE_LOCK_NUMBER

Reply:
Done